match/lib/match/storage/google_cloud_storage.rb

```diff
@@ -0,0 +1,245 @@
+require 'fastlane_core/command_executor'
+require 'fastlane_core/configuration/configuration'
+require 'google/cloud/storage'
+
+require_relative '../options'
+require_relative '../module'
+require_relative './interface'
+
+module Match
+  module Storage
+    # Store the code signing identities in on Google Cloud Storage
+    class GoogleCloudStorage < Interface
+      DEFAULT_KEYS_FILE_NAME = "gc_keys.json"
+
+      # User provided values
+      attr_accessor :type
+      attr_accessor :platform
+      attr_accessor :bucket_name
+      attr_accessor :google_cloud_keys_file
+
+      # Managed values
+      attr_accessor :gc_storage
+
+      # Append Google Cloud specific options to `Match::Options`
+      Match::Options.append_option(
+        FastlaneCore::ConfigItem.new(
+          key: :google_cloud_bucket_name,
+          env_name: "MATCH_GOOGLE_CLOUD_BUCKET_NAME",
+          description: "Name of the Google Cloud Storage bucket to use",
+          optional: true
+        )
+      )
+      Match::Options.append_option(
+        FastlaneCore::ConfigItem.new(
+          key: :google_cloud_keys_file,
+          env_name: "MATCH_GOOGLE_CLOUD_KEYS_FILE",
+          description: "Path to the `#{DEFAULT_KEYS_FILE_NAME}` file",
+          optional: true,
+          verify_block: proc do |value|
+            UI.user_error!("Could not find keys file at path '#{File.expand_path(value)}'") unless File.exist?(value)
+          end
+        )
+      )
+
+      def self.configure(params)
+        if params[:git_url].to_s.length > 0
+          UI.important("Looks like you still define a `git_url` somewhere, even though")
+          UI.important("you use Google Cloud Storage. You can remove the `git_url`")
+          UI.important("from your Matchfile and Fastfile")
+          UI.message("The above is just a warning, fastlane will continue as usual now...")
+        end
+
+        return self.new(
+          type: params[:type].to_s,
+          platform: params[:platform].to_s,
+          google_cloud_bucket_name: params[:google_cloud_bucket_name],
+          google_cloud_keys_file: params[:google_cloud_keys_file]
+        )
+      end
+
+      def initialize(type: nil,
+                     platform: nil,
+                     google_cloud_bucket_name: nil,
+                     google_cloud_keys_file: nil)
+        self.type = type if type
+        self.platform = platform if platform
+        self.google_cloud_keys_file = google_cloud_keys_file
+        self.bucket_name = google_cloud_bucket_name
+
+        ensure_keys_file_exists
```

Comment:
avoid functions that have mysterious side-effects (no inputs, no outputs).

I would recommend doing something like:
```ruby
this.google_cloud_keys_file = ensure_keys_file_exists
```